src/Analysis/Ast/Impl/Analyzer/Symbols/FunctionEvaluator.cs

```diff
@@ -79,6 +79,10 @@ public FunctionEvaluator(ExpressionEval eval, PythonFunctionOverload overload)
                 if (ctor || annotationType.IsUnknown() || Module.ModuleType == ModuleType.User) {
                     // Return type from the annotation is sufficient for libraries and stubs, no need to walk the body.
                     FunctionDefinition.Body?.Walk(this);
+                    // For libraries remove declared local function variables to free up some memory.
+                    if (Module.ModuleType != ModuleType.User) {
```

Comment:
Will this break goto-def navigation?

Reply:
Yes, inside code of the function in a library. Should not affect navigation to a function but navigation and other functionality on variables declared within the function - yes. I am wondering if this should be an option and how to name it then.